components/api/src/main/kotlin/com/hotels/styx/api/MeterRegistry.kt

```diff
@@ -0,0 +1,216 @@
+/*
+  Copyright (C) 2013-2021 Expedia Inc.
+
+  Licensed under the Apache License, Version 2.0 (the "License");
+  you may not use this file except in compliance with the License.
+  You may obtain a copy of the License at
+
+  http://www.apache.org/licenses/LICENSE-2.0
+
+  Unless required by applicable law or agreed to in writing, software
+  distributed under the License is distributed on an "AS IS" BASIS,
+  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+  See the License for the specific language governing permissions and
+  limitations under the License.
+ */
+package com.hotels.styx.api
+
+import io.micrometer.core.instrument.*
+import io.micrometer.core.instrument.MeterRegistry.Config
+import io.micrometer.core.instrument.search.RequiredSearch
+import io.micrometer.core.instrument.search.Search
+import java.util.concurrent.TimeUnit
+import java.util.function.Consumer
+import java.util.function.ToDoubleFunction
+import java.util.function.ToLongFunction
+
+interface MeterRegistry {
```

Comment:
Possibly this interface could be named to include information about what extra functionality it adds to the Micrometer MeterRegistry?
So maybe XxxMeterRegistry.  Just an idea :)

Reply:
It doesn't really add new functionality, besides the ability to create a `ScopedMeterRegistry` from it. The main improvement is that it is an interface rather than a class with internal behaviour.
From Styx's PoV, it is a general purpose meter registry, so adding things to the name would be misleading.

Reply:
👍